apps/src/code-studio/pd/workshop_enrollment/WorkshopEnrollmentCelebrationDialog.jsx

```diff
@@ -2,52 +2,262 @@ import PropTypes from 'prop-types';
+  session,
+  workshopTitle,
+  workshopLocation
+) => {
+  const date = `${session.year}${session.month}${session.day}`;
+  const startTime = `${date}T${session.start_hour}${session.start_min}00`;
+  const endTime = `${date}T${session.end_hour}${session.end_min}00`;
+
+  return `https://calendar.google.com/calendar/render?action=TEMPLATE&text=${encodeURIComponent(
+    workshopTitle
+  )}&location=${encodeURIComponent(
+    workshopLocation
+  )}&dates=${startTime}/${endTime}`;
+};
+
+export const buildOutlookCalendarLink = (
+  session,
+  workshopTitle,
+  workshopLocation
+) => {
+  const date = `${session.year}-${session.month}-${session.day}`;
+  const startTime = `${date}T${session.start_hour}:${session.start_min}:00`;
+  const endTime = `${date}T${session.end_hour}:${session.end_min}:00`;
+
+  return `https://outlook.live.com/calendar/action/compose?rru=addevent&subject=${encodeURIComponent(
+    workshopTitle
+  )}&location=${encodeURIComponent(
+    workshopLocation
+  )}&startdt=${startTime}&enddt=${endTime}`;
+};
+
 export default function WorkshopEnrollmentCelebrationDialog({
   workshopTitle,
+  workshopLocation,
+  workshopSessionInfo,
   onClose,
 }) {
+  const hasMultipleSessions =
+    workshopSessionInfo && workshopSessionInfo.length > 1;
   const [isOpen, setIsOpen] = useState(true);
-  const onCloseDialog = () => {
+  const [multipleSessionDialogType, setMultipleSessionDialogType] =
+    useState('');
+
+  const onCloseCelebrationDialog = () => {
     if (onClose) {
       onClose();
     }
     setIsOpen(false);
   };
 
+  const onCloseSessionCalendarDialog = () => {
+    setMultipleSessionDialogType('');
+    onCloseCelebrationDialog();
+  };
+
+  const getCalendarLink = (session, calendarType) => {
+    if (calendarType === 'Google') {
+      return buildGoogleCalendarLink(session, workshopTitle, workshopLocation);
+    } else if (calendarType === 'Outlook') {
+      return buildOutlookCalendarLink(session, workshopTitle, workshopLocation);
+    }
+  };
+
+  const RenderCalendarSessionDialog = () => {
+    return (
+      <AccessibleDialog
+        className={style.celebrationContainer}
+        onClose={() => setMultipleSessionDialogType('')}
+        closeOnClickBackdrop={true}
+      >
+        <div className={style.showMultipleSessionDialogContainer}>
+          <Heading3>{i18n.enrollmentCelebrationAddToCalendarTitle()}</Heading3>
+          <hr />
+          <BodyTwoText>
+            {i18n.enrollmentCelebrationAddToCalendarDesc()}
+          </BodyTwoText>
+          <table>
+            <thead>
+              <tr>
+                <th>
+                  <Heading6>{i18n.date()}</Heading6>
+                </th>
+                <th>
+                  <Heading6>{i18n.time()}</Heading6>
+                </th>
+                <th />
+              </tr>
+            </thead>
+            <tbody>
+              {workshopSessionInfo.map((session, index) => (
+                <tr key={`session-${index}`}>
```

Comment:
nit: does session have a unique identifier you could use here? It's not best practice to use the index as the key, though I think in this case it doesn't matter (these rows wouldn't be reordered or anything)

Reply:
Oh for sure!